### ISSUE TITLE
Changing screens doesn't work

### ISSUE THREAD

**Issue by thaprozz** [react-navigation/react-navigation#5042]
## Description

I'm trying to navigate to another screen (Artist) by clicking on an element in a FlatList. This FlatList contains Artist instances, as set in the _renderListItem method. 

I managed to pass the params to the other screen, but unfortunately the navigation itself doesn't seem to work; I'm getting the expected values in my logs, but nothing happens and the app stays at LinksScreen (doesn't change screens).

## Reproducible Demo

https://snack.expo.io/@jvdl2711/artist-navigation-issue

**LinksScreen.js**

```
export default class LinksScreen extends React.Component {
  constructor(props) {
    super(props);
    this._onAlertTypePressed = this._onAlertTypePressed.bind(this);
  }

  _onAlertTypePressed(typeId: string, typeName: string, imageUrl: string) {
    console.log(typeId)
    console.log(typeName)
    console.log(imageUrl)

// NavigationActions -> I used to try it this way, but this results in the same thing: screen doens't change
    // NavigationActions.navigate({
    //                 routeName: 'Artist',
    //                 params: { itemId: typeId, itemName: typeName, itemImageUrl: imageUrl,},
    //               });

// NAVIGATION
     this.props.navigation.navigate('HomeStack',{},
     {
       type: "Navigate",
       routeName: "Artist",
       params: {
         itemId: typeId,
         itemName: typeName,
         itemImageUrl: imageUrl}
             }
           );

// This method is called in the FlatList to render individual Artist objects for the list in LinksScreen
  _renderListItem = ({item}) => (
    <Artist
      itemId={item.id}
      itemName={item.title.rendered}
      itemImageUrl={
        item.better_featured_image
          ? item.better_featured_image.source_url
          : 'http://54.168.73.151/wp-content/uploads/2018/04/brand-logo.jpg'
      }
      onPressItem={this._onAlertTypePressed}
    />
  );

// Not used anymore, but maybe it's better not to render the whole object and do everything using this method. Unfortunately I didn't manage to solve it this way either, so I tried the approach found above.
  renderRow = item => (
    <View style={styles.grid}>
      <Image
        style={styles.thumb}
        source={{
          uri: item.better_featured_image
            ? item.better_featured_image.source_url
            : 'http://54.168.73.151/wp-content/uploads/2018/04/brand-logo.jpg',
        }}
      />
      <Text style={styles.title}>{item.title.rendered}</Text>
    </View>
  );

  getKey = item => String(item.id);

// This is where the component gets rendered (called in render())
  renderComponent() {
    if (this.state.isLoading) {
      return <ActivityIndicator />;
    } else if (this.state.isError) {
      return <Text>Error loading data</Text>;
    } else {
      return (
        <FlatList
          numColumns={3}
          contentContainerStyle={styles.elementsContainer}
          data={this.state.data}
          renderItem={this._renderListItem}
          keyExtractor={this.getKey}
        />
      );
    }
  }

  render() {
    return (
      <View style={styles.container}>
        <Text
          style={{
            fontSize: 20,
            color: '#FFFFFF',
            marginLeft: 4,
            marginTop: 10,
          }}>
          RESIDENTS
        </Text>
        {this.renderComponent()}
      </View>
    );
  }
}

```
**Artist.js**

The console.log in the beginning of the _onPress() method seem to be working (the expected params have the correct values here), but I'm unable to navigate to this screen.

```
class Artist extends React.PureComponent {

//This method gets called when pressing an item in the LinksScreen. The params have the expected values.
  _onPress = () => {
     // console.log(this.props)

     const itemId = this.props.itemId
     const itemName = this.props.itemName
     const itemImageUrl = this.props.itemImageUrl

  render() {
    return (
      <TouchableOpacity
        {...this.props}
        style={styles.grid}
        onPress={this._onPress}>
        <Image
          style={styles.image}
          source={{uri: this.props.itemImageUrl}}
        />
        <Text style={styles.title}>{this.props.itemName}</Text>
      </TouchableOpacity>
    );
  }
}
export default Artist;
```

**MainTabNavigator.js**

Perhaps there is something wrong regarding the routing, so here is how it's done in my case.

```
const HomeStack = createStackNavigator({
  Home: {
    screen: HomeScreen,
  },
  Links: {
    screen: LinksScreen,
  },
  Artist: {
    screen: Artist,
  },
});

AppRegistry.registerComponent('ParamsRepo', () => HomeStack);

export default HomeStack;
```

## Environment

[skip envinfo]
  "dependencies": {
    "@expo/samples": "2.1.1",
    "axios": "^0.18.0",
    "expo": "^30.0.1",
    "react": "16.3.1",
    "react-native": "https://github.com/expo/react-native/archive/sdk-30.0.0.tar.gz",
    "react-native-htmlview": "^0.13.0",
    "react-navigation": "^2.16.0",
    "react-redux": "^5.0.7",
    "redux": "^4.0.0",
    "redux-thunk": "^2.3.0"
  },
  "devDependencies": {
    "jest-expo": "30.0.0"
  }
}

**Comment by brentvatne** [comment]
I helped you with this only because you seem to really be having a hard time, these issues are reserved for problems that you can demonstrate are bugs in react-navigation itself and not in your own code (like this issue). Read the code in https://snack.expo.io/@notbrent/artist-navigation-issue to see this working. Notice that you need to get params off of the navigation object.

I think it's important that you take a step back and read the documentation at https://reactnavigation.org/ before proceeding further. Any additional questions you might have that you can't demonstrate are bugs in React Navigation belong on StackOverflow.

**Comment by thaprozz** [comment]
Hey Brent!

You're definitely a great person. I was asking my question at the wrong place and you still helped me, even if you don't need to. I appreciate that a lot, man :) 

I've been reading through the code and I managed to simplify things a little bit, fortunately (not that you made it complicated, but I did). Check [https://snack.expo.io/@jvdl2711/artist-navigation-issue](url) if you're interested, because of course it's based on your help!

It's my first time using React Native, but I'll absolutely take a look at the basics again before moving on. I just wanted this to be done out of frustration. Again, I appreciate it a lot and have an amazing day.
